lenskit-core/src/main/java/org/lenskit/data/entities/AbstractEntity.java

```diff
@@ -57,6 +57,16 @@ public EntityType getType() {
     /**
      * {@inheritDoc}
      *
+     * This implementation delegates to {@link #hasAttribute(String)}
+     */
+    @Override
+    public boolean hasAttribute(TypedName<?> name) {
+        return hasAttribute(name.getName());
```

Comment:
Hmm this is compatible with API. That said the API has no clean way of verifying that an entity has an attribute of the correct name and type. Most of the time that shouldn't be a problem but I could imagine wanting to duck type entities from time to time.